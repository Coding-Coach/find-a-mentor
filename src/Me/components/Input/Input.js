import React, { useContext } from 'react';
import styled from 'styled-components';
import { formFieldContext } from '../FormField/formContext';

const StyledInput = styled.input`
  font-family: Lato, sans-serif;
  font-size: 14px;
  line-height: 17px;
  border-radius: 3px;
  background-color: #fff;
  border: 1px solid #bfbfbf;
  padding: 7px 12px 6px 8px;
  color: #4f4f4f;
  ::placeholder {
    color: #898889;
  }
  :disabled {
    background-color: #dadada;
  }
`;

export const Input = props => {
  const id = useContext(formFieldContext);
  return <StyledInput id={id} {...props} />;
};
