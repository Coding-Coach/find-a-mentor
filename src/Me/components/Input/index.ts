export { Input as default } from './Input';
