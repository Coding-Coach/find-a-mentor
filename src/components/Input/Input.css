.label {
  position: absolute;
  top: -10px;
  left: 25px;
  background-color: #fff;
  text-transform: uppercase;
  color: #4a4a4a;
  font-weight: bold;
}

.input-container {
  position: relative;
  margin-top: 20px;
}

.input {
  width: 250px;
  height: 40px;
  background: none;
  border: none;
  border: 1px solid rgba(155, 155, 155, 0.5);
  border-radius: 2px;
  margin-left: var(--filter-input-offset);
  padding-left: 10px;
  color: var(--tag-color);
}

.input.input-extended {
  width: 100%;
}

@media all and (max-width: 800px) {
  .input {
    width: calc(100% - var(--filter-input-offset) * 2);
    font-size: 16px;
  }
}