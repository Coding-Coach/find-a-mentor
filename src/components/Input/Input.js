import React from "react";

import "./Input.css";

const Input = ({ id, label, children }) => (
  <div className="input-container">
    <label htmlFor={id} className="label">
      {label}
    </label>
    {children}
  </div>
);

export default Input;
