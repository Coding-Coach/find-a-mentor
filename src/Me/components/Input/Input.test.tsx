import React from 'react';
import Input from './';
import { render } from '@testing-library/react';

describe('Input component', () => {
  it('renders properly', () => {
    const { container } = render(<Input placeholder="test" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
