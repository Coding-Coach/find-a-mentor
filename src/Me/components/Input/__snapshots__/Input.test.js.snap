// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders properly 1`] = `
.c0 {
  font-family: Lato,sans-serif;
  font-size: 14px;
  line-height: 17px;
  border-radius: 3px;
  background-color: #fff;
  border: 1px solid #bfbfbf;
  padding: 7px 12px 6px 8px;
  color: #4f4f4f;
}

.c0::-webkit-input-placeholder {
  color: #898889;
}

.c0::-moz-placeholder {
  color: #898889;
}

.c0:-ms-input-placeholder {
  color: #898889;
}

.c0::placeholder {
  color: #898889;
}

.c0:disabled {
  background-color: #dadada;
}

<input
  class="c0"
  placeholder="test"
/>
`;
